format.go

```diff
@@ -161,18 +159,29 @@ func (df *file) doFmt(ast *parser.Node) (result string, err error) {
 		v = fmtCopy(ast.Next)
 	}
 
-	// print to the result
-	result = fmt.Sprintf("%s\t%s\n", k, v)
+	// print line to the result
+	result += fmt.Sprintf("%s\t%s\n", k, v)
 
-	// set our current line as the start line in the next node
-	// since we want the next node
-	df.currentLine++
+	// move forward
 	if ast.Next != nil {
-		df.currentLine = ast.Next.StartLine
+		df.currentLine = ast.StartLine
 	}
 	return
 }
 
+func (df *file) getCommentsUntil(until int) (result string, err error) {
+	comments, err := df.getOriginalLines(df.currentLine, until, df.name)
+	if err != nil {
+		return "", err
+	}
+
+	re := regexp.MustCompile(`#.*\n`)
```

Comment:
this might be an issue for windows but enough for our purposes.

Reply:
how come ?

Reply:
line ending is `\r\n` but you're appending `\n` later.